rdar://109697000 (`Scripts/generate-serializers.py` produces non-deterministic outputs)

aproskuryakov opened the pull request:
#### 9c21df951f6109ccdf9f86620c6ad76e4a07946f
<pre>
rdar://109697000 (`Scripts/generate-serializers.py` produces non-deterministic outputs)

Reviewed by Sihui Liu and Elliott Williams.

Patch by Argyrios Kyrtzidis.

* Source/WebKit/Scripts/generate-serializers.py:
(ConditionalHeader.__lt__):
(ConditionalHeader.__lt__.condition_str):

Canonical link: <a href="https://commits.webkit.org/264604@main">https://commits.webkit.org/264604@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cd7cd8f2d8467a2e082da952a650981c78995564

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8064 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/8349 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8562 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9716 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8165 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/10335 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/8260 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11023 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/8208 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/9286 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7350 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9840 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/8194 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6587 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/14969 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7710 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/7530 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/10874 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7975 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6490 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7288 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1944 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11497 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7716 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple reviewed (approved):
r+, though I think you need to rewrite the commit message in WebKit style.

szewai reviewed (approved):
r=me, can you mention why we want this change in commit message?

aproskuryakov committed:
rdar://109697000 (`Scripts/generate-serializers.py` produces non-deterministic outputs)

Reviewed by Sihui Liu and Elliott Williams.

Patch by Argyrios Kyrtzidis.

* Source/WebKit/Scripts/generate-serializers.py:
(ConditionalHeader.__lt__):
(ConditionalHeader.__lt__.condition_str):

Canonical link: https://commits.webkit.org/264604@main
